bundles/core/src/main/java/com/adobe/cq/commerce/core/search/internal/services/SearchResultsServiceImpl.java

```diff
@@ -109,23 +113,23 @@ public SearchResultsSet performSearch(
 
         if (magentoGraphqlClient == null) {
             LOGGER.error("The search result service was unable to create a new MagentoGraphqlClient.");
-            return searchResultsSet;
+            return new ImmutablePair<>(null, searchResultsSet);
         }
 
         // We will use the search filter service to retrieve all of the potential available filters the commerce system
         // has available for querying against
         List<FilterAttributeMetadata> availableFilters = searchFilterService.retrieveCurrentlyAvailableCommerceFilters(page);
 
         // Next we generate the graphql query and actually query the commerce system
-        String queryString = generateQueryString(searchOptions, availableFilters, productQueryHook);
+        String queryString = generateQueryString(searchOptions, availableFilters, productQueryHook, categoryRetriever);
         LOGGER.debug("Generated query string {}", queryString);
         GraphqlResponse<Query, Error> response = magentoGraphqlClient.execute(queryString);
 
         // If we have any errors returned we'll log them and return an empty search result
         if (response.getErrors() != null && response.getErrors().size() > 0) {
             response.getErrors().stream()
                 .forEach(err -> LOGGER.error("An error has occurred: {} ({})", err.getMessage(), err.getCategory()));
-            return searchResultsSet;
+            return new ImmutablePair<>(null, searchResultsSet);
```

Comment:
I think here you could still try to return the category (of course if there is any `data`), the errors might be related to something else. GraphQL can return data and errors together.